lib/Dialect/FIRRTL/Transforms/InferWidths.cpp

```diff
@@ -1518,6 +1518,17 @@ LogicalResult InferenceMapping::mapOperation(Operation *op) {
         }
       })
 
+      .Case<RefSendOp>([&](auto op) {
+        declareVars(op.getResult(), op.getLoc());
+        constrainTypes(op.getResult(), op.getBase());
+      })
+      .Case<RefResolveOp>([&](auto op) {
+        declareVars(op.getResult(), op.getLoc());
+        // Both directions, resolve may flow either way.
+        unifyTypes(FieldRef(op.getResult(), 0), FieldRef(op.getRef(), 0),
+                   op.getResult().getType().template cast<FIRRTLType>());
+      })
```

Comment:
I've anecdotally had problems with MSVC having issues with the later use of `template` construction. I tend to prefer making the `auto` an explicit type to avoid the need for the template:

```suggestion
      .Case<RefResolveOp>([&](RefResolveOp op) {
        declareVars(op.getResult(), op.getLoc());
        // Both directions, resolve may flow either way.
        unifyTypes(FieldRef(op.getResult(), 0), FieldRef(op.getRef(), 0),
                   op.getResult().getType().cast<FIRRTLType>());
      })
```

Also: The comment makes me slightly nervous. This won't enable inference of an unknown width `ref.send` from a known widht on the `ref.resolve`, yes?

Reply:
(and touched up to avoid need for the `template` :+1:)